### PR TITLE
[telemundo] add extractor (#284)

### PR DESCRIPTION
**Pull request by king-millez**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

---

This PR addresses #284 by adding an info extractor for Telemundo.

### CODE REVIEW

File: yt_dlp/extractor/telemundo.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        webpage = self._download_webpage(url, video_id)
+        metadata = json.loads(
+            self._search_regex(r'<[^>]+id="__NEXT_DATA__"[^>]+>([^<]+)', webpage, 'JSON metadata'))
+        redirect_url = try_get(
+            metadata,
+            lambda x: x['props']['initialState']['video']['associatedPlaylists'][0]['videos'][0]['videoAssets'][0]['publicUrl'])
+
+        m3u8_url = self._request_webpage(HEADRequest(
+            redirect_url + '?format=redirect&manifest=m3u&format=redirect&Tracking=true&Embedded=true&formats=MPEG4'),
+            video_id, 'Processing m3u8').geturl()
```
Comment (pukkandan) [inline yt_dlp/extractor/telemundo.py:47]: is this needed?

File: yt_dlp/extractor/telemundo.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        }
+    }, {
+        'url': 'https://www.telemundo.com/shows/al-rojo-vivo/empleo/video/personajes-de-times-square-piden-que-la-ciudad-de-nueva-york-los-deje-volver-trabajar-tmvo9816272',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        metadata = json.loads(
```
Comment (pukkandan) [inline yt_dlp/extractor/telemundo.py:39]: use `self._parse_json`. You also might need to use `transform_source=js_to_json` if you are extracting from js and not json

File: yt_dlp/extractor/telemundo.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        date = unified_timestamp(try_get(
+            metadata, lambda x: x['props']['initialState']['video']['associatedPlaylists'][0]['videos'][0]['datePublished'].split(' ', 1)[1]))
+        return {
+            'url': url,
+            'id': video_id,
+            'title': self._search_regex(r'<h1[^>]+>([^<]+)', webpage, 'title'),
+            'formats': formats,
+            'timestamp': date,
+            'uploader': 'Telemundo',
+            'uploader_id': self._search_regex(r'https?:\/\/(?:[^/]+\/){3}video\/(?P<id>[^\/]+)', m3u8_url, 'Akamai account')
```
Comment (pukkandan) [inline yt_dlp/extractor/telemundo.py:59]: `fatal=False`

File: yt_dlp/extractor/telemundo.py (hunk 1 above)
Comment (king-millez) [reply yt_dlp/extractor/telemundo.py:47]: Yep, gets the Akamai account. If that changes, we wouldn't know from the original URL.

File: yt_dlp/extractor/telemundo.py (hunk 2 above)
Comment (king-millez) [reply yt_dlp/extractor/telemundo.py:39]: The `script`'s type is `application/json`, so `js_to_json` isn't needed.

File: yt_dlp/extractor/telemundo.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/telemundo.py:47]: ok, gotcha

File: yt_dlp/extractor/telemundo.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+from ..utils import (
+    try_get,
+    HEADRequest,
+    unified_timestamp
+)
+
+
+class TelemundoIE(InfoExtractor):
+
+    _VALID_URL = r'https?:\/\/(?:www\.)?telemundo\.com\/.+?(?=video)video\/[^\/]+(?P<id>tmvo\d{7})'
```
Comment (pukkandan) [inline yt_dlp/extractor/telemundo.py:14]: why `(?=video)`?

Comment (pukkandan) [reply yt_dlp/extractor/telemundo.py:14]:
```suggestion
    _VALID_URL = r'https?:\/\/(?:www\.)?telemundo\.com\/.+?video\/[^\/]+(?P<id>tmvo\d{7})'
```

Hey, you never replied...

**Pull request merged**